csrc/device_lower/pass/replace_size.cpp

```diff
@@ -230,20 +192,28 @@ void replaceSymbolicSizes(Fusion* fusion) {
 
   // Use a minimal number of sizes from provided tensors.
   auto extent_simplification_map = getSimplificationMap(fusion);
+
+  // We now need to map replacement scalars to their targets in tensor_dim_map
+  // if they exist. To do this we compose extent_simplification_map with
+  // tensor_dim_map.
   for (auto extent_entry : extent_simplification_map) {
     auto orig_extent = extent_entry.first;
     auto simplified_extent = extent_entry.second;
-    if (tensor_dim_map.count(orig_extent)) {
-      if (tensor_dim_map.count(simplified_extent)) {
-        tensor_dim_map[orig_extent] = tensor_dim_map[simplified_extent];
-      } else {
-        tensor_dim_map[orig_extent] = simplified_extent;
-      }
+    auto it = tensor_dim_map.find(simplified_extent);
+    if (it != tensor_dim_map.end()) {
```

Comment:
Why are we doing this? Previously, there's `tensor_dim_map`, and we update it with `expr_simplification_map`. Now, it seems we are doing the opposite, which means if an extent is discovered to be a constant in `expr_simplification_map`, it would be overwritten by the symbolic representation. Am I understanding it correctly?

Reply:
Yes, we are now doing more replacements than previously. Previously we would only replace extents that are exact mapped to input extents. This was a problem in the motivating example from #2702: see https://github.com/NVIDIA/Fuser/issues/2702#issuecomment-2256920406. In that case it's "`ceilDiv`" extent of a reshaped tensor that needs to be mapped as constant and it's not actually exact mapped with an input.

Reply:
I understand that, but what happens if an input domain is discovered to be a constant in `extent_simplification_map`. Wouldn't it be overwritten by the `getMetaData` expr even if it's constant?

Reply:
Ah, I see.

Why is the previous code not enough?

Reply:
Hmm, what about line 240 of the previous code? Doesn't it add a mapping that does not exist in `tensor_dim_map`?

Reply:
Never mind, I misread the code. It's under the `if` branch starting at line 237, not 236.